packages/flutter/test/widgets/sliver_fill_viewport_test.dart

```diff
@@ -145,4 +146,50 @@ void main() {
       ),
     );
   });
+
+  testWidgets('SliverFillViewport padding test', (WidgetTester tester) async {
+    final SliverChildListDelegate delegate = SliverChildListDelegate(
+      <Widget>[
+        Container(child: const Text('0')),
+      ],
+      addAutomaticKeepAlives: false,
+      addSemanticIndexes: false,
+    );
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: CustomScrollView(
+          slivers: <Widget>[
+            SliverFillViewport(
+              padEnds: true,
+              viewportFraction: 0.5,
+              delegate: delegate,
+            ),
+          ],
+        ),
+      ),
+    );
+
+    final RenderSliver boxWithPadding = tester.renderObject<RenderSliver>(find.byType(SliverFillViewport).first);
```

Comment:
nit: is there more than one `SliverFillViewport` widget? I think removing the `.first` should be fine? 
```suggestion
    final RenderSliver boxWithPadding = tester.renderObject<RenderSliver>(find.byType(SliverFillViewport));
```

Reply:
Didn't know you could do that :P